src/CommonITILTask.php

```diff
@@ -1281,7 +1281,11 @@ public static function genericPopulatePlanning($itemtype, $options = [])
 
                       /// Specific for tickets
                         $interv[$key]["device"] = [];
-                        if (isset($parentitem->hardwaredatas) && !empty($parentitem->hardwaredatas)) {
+                        if (
+                            $parentitem instanceof Ticket
```

Comment:
```suggestion
                            $parentitem instanceof CommonITILObject
```

no ?

Reply:
The error was:
```
[2023-10-04 13:29:59] glpiphplog.WARNING:   *** PHP User Warning (512): Unknown field: 'hardwaredatas' in D:\Datas\www\glpi\src\CommonITILObject.php at line 269
  Backtrace :
  src\CommonITILObject.php:269                       trigger_error()
  src\CommonITILTask.php:1284                        CommonITILObject->__isset()
  src\ChangeTask.php:166                             CommonITILTask::genericPopulatePlanning()
  src\Planning.php:2236                              ChangeTask::populatePlanning()
  src\Planning.php:2005                              Planning::constructEventsArraySingleLine()
  ajax\planning.php:50                               Planning::constructEventsArray()
  public\index.php:82                                require()
```

![image](https://github.com/glpi-project/glpi/assets/8530352/e2fc8a6e-32d4-4dbd-8b1e-14fbba845809)

`hardwaredatas` is only used for tickets, I don't know if it's relevant for change or problem?

Reply:
ok fine